packages/office-ui-fabric-react/src/components/Dialog/Dialog.base.tsx

```diff
@@ -123,33 +125,36 @@ export class DialogBase extends React.Component<IDialogProps, {}> {
       ...this.props.dialogContentProps,
       draggableHeaderClassName: dialogDraggableClassName,
       titleProps: {
+        // tslint:disable-next-line:deprecation
         id: this.props.dialogContentProps?.titleId || this._defaultTitleTextId,
         ...this.props.dialogContentProps?.titleProps
       }
     };
 
     const classNames = getClassNames(styles!, {
       theme: theme!,
-      className: className || mergedModalProps.className,
-      containerClassName: containerClassName || mergedModalProps.containerClassName,
+      className: mergedModalProps.className || className,
+      containerClassName: mergedModalProps.containerClassName || containerClassName,
       hidden,
       dialogDefaultMinWidth: minWidth,
       dialogDefaultMaxWidth: maxWidth
     });
 
+    const isModalBlocking = mergedModalProps.isBlocking !== undefined ? mergedModalProps.isBlocking : isBlocking;
+
     return (
       <Modal
         elementToFocusOnDismiss={elementToFocusOnDismiss}
         firstFocusableSelector={firstFocusableSelector}
         forceFocusInsideTrap={forceFocusInsideTrap}
         ignoreExternalFocusing={ignoreExternalFocusing}
         isClickableOutsideFocusTrap={isClickableOutsideFocusTrap}
-        onDismissed={onDismissed}
+        onDismissed={mergedModalProps.onDismissed !== undefined ? mergedModalProps.onDismissed : onDismissed}
```

Comment:
Can you not just include these defaults with the mergedModalProps?

Reply:
Good catch! Done!